app/com/m3/octoparts/util/OctoMetricsImpl.scala

```diff
@@ -0,0 +1,27 @@
+package com.m3.octoparts.util
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.kenshoo.play.metrics.MetricsImpl
+import play.api.{ Logger, Configuration }
+import play.api.inject.ApplicationLifecycle
+
+import scala.util.control.NonFatal
+
+/**
+ * During tests, we occasionally have cases where the asynchronous shutting down of an application means that
+ * the previous registry in the singleton has not been unregistered before starting the next test.
+ *
+ * This implementation of Metrics overrides onStart to perform onStop cleanups first before proceeding with
+ * other onStart duties
+ */
+class OctoMetricsImpl(lifecycle: ApplicationLifecycle, configuration: Configuration) extends MetricsImpl(lifecycle, configuration) {
+
+  override def onStart(): ObjectMapper = {
```

Comment:
Calling `onStart` attempts to remove the metric from the internal map. I'm guessing the error thrown would be from the concurrent map [in use here in Codahale Metrics](https://github.com/dropwizard/metrics/blob/d8d8e2796d1f7dfb5268e2e8fbf35039570c53d7/metrics-core/src/main/java/io/dropwizard/metrics/SharedMetricRegistries.java#L26) itself [called from Kenshoo Metrics](https://github.com/kenshoo/metrics-play/blob/master/src/main/scala/com/kenshoo/play/metrics/Metrics.scala#L80). What would it throw specifically, that it could not remove a key not in the map?